test/jdk/java/nio/file/Files/probeContentType/Basic.java

```diff
@@ -188,9 +189,10 @@ public static void main(String[] args) throws IOException {
         exTypes.add(new ExType("xlsx", List.of("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")));
         exTypes.add(new ExType("wasm", List.of("application/wasm")));
 
-        // extensions with content type that differs on Windows 11+
+        // extensions with content type that differs on Windows 11+ and
+        // Windows Server 2025
         if (OperatingSystem.isWindows() &&
-            (System.getProperty("os.name").endsWith("11") ||
+            (StaticProperty.osName().matches("^.*[11|2025]$") ||
```

Comment:
Hello Brian, the change looks reasonable to me. If we want to be extra "strict" on which Windows version this conditional is applicable for, then maybe we should include the regex pattern for one or more whitespace before the the `[11|2025]`. But given what this test is for, I think it's fine in this proposed form too.

Reply:
> maybe we should include the regex pattern for one or more whitespace before the the `[11|2025]`. But given what this test is for, I think it's fine in this proposed form too.

Hi @jaikiran. Thanks for the prompt review. I'll give it some more thought tomorrow, but might just leave it as is.

Reply:
> I'll give it some more thought tomorrow, but might just leave it as is.

Given the newly filed issue [JDK-8347286](https://bugs.openjdk.org/browse/JDK-8347286), I think I'll leave this suggestion until later, as it might simply vanish.